apps/prairielearn/src/pages/partials/questionFooter.ejs

```diff
@@ -21,7 +21,13 @@
             </button>
           <% } %>
           <% if (showGradeButton) { %>
-            <button class="btn btn-primary question-grade disable-on-submit order-1 mr-1" <%= disableGradeButton ? "disabled" : "" %> name="__action" value="grade">Save & Grade
+            <%
+            let invalidRoleConfig = false;
+            if (typeof groupConfig !== 'undefined' && groupConfig.has_roles && typeof rolesInfo !== 'undefined') {
+              invalidRoleConfig = rolesInfo.validationErrors.length > 0 || !rolesInfo.rolesAreBalanced || rolesInfo.usersWithoutRoles.length > 0;
+            }
+            %>
+            <button class="btn btn-primary question-grade disable-on-submit order-1 mr-1" <%= disableGradeButton || invalidRoleConfig ? "disabled" : "" %> name="__action" value="grade">Save & Grade
```

Comment:
Is it intentional that students without the "submit" permission can still save answers? For exam assessments, all saved answers will be graded when the assessment closes, so it seems like a student without "submit" permissions can still in effect cause a submission to be created and graded.

Reply:
Keeping "save" enabled was intentional, since we had decided it may be best for collaboration to allow multiple students to save potential answers while only one student could hit "submit", but I see your point with this being an issue in exams where "save" is basically akin to submitting if the group only submits once at the end.

I suppose if we wanted to give instructors the most flexibility, we could make another question-level permission `can_save` (basically duplicating `can_submit` everywhere in the code). @renzol2 what do you think? We could also ask the SIIP team again.

Reply:
Looks great! Thanks for following up with Geoffrey/Mariana about that.